### PR TITLE
i forgot what this patch does

### PR DESCRIPTION
**Pull request by uwx**

### CODE REVIEW

File: DSharpPlus/Entities/DiscordUser.cs [hunk 1]
```diff
@@ -87,6 +88,21 @@ internal DiscordUser(TransportUser transport)
         public string Mention => Formatter.Mention(this, this is DiscordMember);
 
         /// <summary>
+        /// Gets whether this user is the Client which created this object.
+        /// </summary>
+        [JsonIgnore]
+        public bool IsCurrentClient => this.Id == this.Discord.CurrentUser.Id;
```
Comment (Emzi0767) [inline DSharpPlus/Entities/DiscordUser.cs]: `IsCurrent` is enough I believe.

File: DSharpPlus/Entities/DiscordMember.cs [hunk 2]
```diff
@@ -268,7 +274,61 @@ public DiscordColor Color
         /// <returns></returns>
         public Task ReplaceRolesAsync(IEnumerable<DiscordRole> roles, string reason = null) =>
             this.Discord.ApiClient.ModifyGuildMemberAsync(this.Guild.Id, this.Id, null, roles.Select(xr => xr.Id), null, null, null, reason);
+
```
Comment (Emzi0767) [inline DSharpPlus/Entities/DiscordMember.cs]: eh... ehhhhhh... not sure about these, ask Naam.

File: DSharpPlus/Entities/DiscordGuild.cs [hunk 3]
```diff
@@ -282,7 +282,7 @@ internal DiscordGuild()
         }
 
         /// <summary>
-        /// Bans a specified member this this guild.
+        /// Bans a specified member from this guild.
```
Comment (NaamloosDT) [inline DSharpPlus/Entities/DiscordGuild.cs]: Sorry just had to comment this made me laugh :^)

File: DSharpPlus/Entities/DiscordMember.cs (hunk 2 above)
Comment (NaamloosDT) [reply DSharpPlus/Entities/DiscordMember.cs]: Seems good to me, honestly. I've originally added the guild object to not have to look up the guild to do some guild-specific operations.

Adding links only makes this easier

File: DSharpPlus/Entities/DiscordUser.cs (hunk 1 above)
Comment (NaamloosDT) [reply DSharpPlus/Entities/DiscordUser.cs]: Please change this to `IsCurrent` ^-^

File: DSharpPlus/DiscordClient.cs [hunk 4]
```diff
@@ -1837,7 +1837,7 @@ internal Task OnResumedAsync()
                 }
                 else
                 {
-                    mentioned_users = Utilities.GetUserMentions(message).Select(xid => this.InternalGetCachedUser(xid)).ToList();
+                    mentioned_users = Utilities.GetUserMentions(message).Select(this.InternalGetCachedUser).ToList();
```
Comment (NaamloosDT) [inline DSharpPlus/DiscordClient.cs]: Never did/see that. Does that actually work?

Comment (uwx) [reply DSharpPlus/DiscordClient.cs]: yes

Comment (uwx) [reply DSharpPlus/DiscordClient.cs]: welcome to java world, we have method references

Comment (Emzi0767) [reply DSharpPlus/DiscordClient.cs]: It's valid is a predicate. It just doesn't grow the callstack as much.

**Comment by NaamloosDT** [comment]
oh shit

seems good now xD

**Comment by NaamloosDT** [comment]
Let's wait for appveyor to complete and emzi to approve c:

File: DSharpPlus/Entities/DiscordMember.cs [hunk 5]
```diff
@@ -268,7 +274,61 @@ public DiscordColor Color
...
+        /// <returns></returns>
+        public Task PlaceInAsync(DiscordChannel channel) =>
+            channel.PlaceMemberAsync(this);
+
+        /// <summary>
+        /// Calculates permissions in a given channel for this member.
+        /// </summary>
+        /// <param name="channel">Channel to calculate permissions for.</param>
+        /// <returns>Calculated permissions for this member in the channel.</returns>
+        public Permissions PermissionsIn(DiscordChannel channel) =>
```
Comment (Emzi0767) [inline DSharpPlus/Entities/DiscordMember.cs]: For consistency, this should be called `PermissionsFor()`

File: DSharpPlus/Entities/DiscordMember.cs [hunk 6]
```diff
@@ -268,7 +274,61 @@ public DiscordColor Color
...
+        public Task RemoveAsync(string reason = null) =>
+            this.Guild.RemoveMemberAsync(this, reason);
+
+        /// <summary>
+        /// Gets audit log entries for this member in their guild.
+        /// </summary>
+        /// <param name="limit">Maximum number of entries to fetch.</param>
+        /// <param name="action_type">Filter by action type.</param>
+        /// <returns>A collection of requested audit log entries.</returns>
+        public Task<IReadOnlyList<DiscordAuditLogEntry>> GetAuditLogsAsync(int? limit = null, AuditLogActionType? action_type = null) =>
```
Comment (Emzi0767) [inline DSharpPlus/Entities/DiscordMember.cs]: I think this should stay on the guild only. The name indicates it gets audit logs from the member, which isn't exactly what his does.

**Pull request merged**